### PR TITLE
Bugfixing and architectural refactor

### PR DESCRIPTION
**Pull request by ecxs-daedalus**
Fix for #22 

- Reordered functions into 3 classes: Probe, Printer and Test_suite
- Changed probe detection to regex to deal with whitespaces
- Added warning if no probe could be detected
- Switched standard output directory top '/tmp' and added a option to change output directory
- Switched several dict reads to getters
- Fixed some sample counting
- Switchd query build to the request libs instead of manual build queries

### CODE REVIEW

File: README.md
Comment (conlank) [inline README.md:1]: One `$HOME` wasn't changed, intentional? Line 62

File: probe_accuracy_test_suite.py [hunk 1]
```diff
@@ -15,540 +15,806 @@
...
-                    userparams["corner"]
-                    or userparams["repeatability"]
-                    or userparams["drift"]
-                    or userparams["speedtest"]
-                ]
-            ):
-                print("Running all tests")
-                userparams.update({"corner": 30, "repeatability": 20, "drift": 100})
+            settings = self.printer.query("configfile", "settings")
+            if settings["docklocation_x"]:
```
Comment (conlank) [inline probe_accuracy_test_suite.py:109]: This doesn't cover the case of using the [dockable_probe] module, although there are starting to be so many versions of the docking macros that makes writing something like this a pain, as there might not be a good way to cover all edge cases elegantly. 

```[dockable_probe]
pin:
z_offset:
sample_retract_dist:
approach_position:
dock_position:
detach_position:
(check_open_attach: OR probe_sense_pin:) AND/OR dock_sense_pin:
```

File: probe_accuracy_test_suite.py [hunk 2]
```diff
@@ -15,540 +15,806 @@
...
+                self.isKlicky = True
+                print(f"{ CLEAR_LINE }Probe type: Klicky mode detected")
+                returnp
```
Comment (conlank) [inline probe_accuracy_test_suite.py:112]: typo

**Review by conlank — APPROVED** [review]
Great job with refactoring this code. Thank you

**Comment by sporkus** [comment]
@ecxs-daedalus, @conlank  
 Thank you for the PR and review. I haven't been keeping up with 3d printing and this project in particular, so my ability to review is quite limited. I am glad this project is still useful for the community.
 
Would it be possible to have usage examples for the new output dir option in the readme?

**Pull request merged**